pkg/api/helpers/errors.go

```diff
@@ -101,19 +102,23 @@ func (e UserVisibleAPIError) Unwrap() error {
 // unsuccessful HTTPResponse field and uses its message, along with a Body
 // that it assumes is an api.Error.
 func ResponseAsError(response interface{}) error {
+	if httpResponse, ok := response.(*http.Response); ok {
+		return HTTPResponseAsError(httpResponse)
+	}
 	r := reflect.Indirect(reflect.ValueOf(response))
 	if !r.IsValid() || r.Kind() != reflect.Struct {
 		return CallFailedError{
 			Message: fmt.Sprintf("bad type %s: must reference a struct", r.Type().Name()),
 			Err:     ErrRequestFailed,
 		}
 	}
-
+	var httpResponse *http.Response
+	var ok bool
 	f := r.FieldByName("HTTPResponse")
 	if !f.IsValid() {
 		return fmt.Errorf("[no HTTPResponse]: %w", ErrRequestFailed)
 	}
-	httpResponse, ok := f.Interface().(*http.Response)
+	httpResponse, ok = f.Interface().(*http.Response)
```

Comment:
remove the above definition - we use them from here
```suggestion
	httpResponse, ok := f.Interface().(*http.Response)
```

Reply:
Agree. Fixed.